### PR TITLE
Add ability for course staff to always view the results.

### PR DESCRIPTION
**Pull request by mtyaka**
Course staff users can now view the results without voting.
The can also see the results for polls/surveys marked as 'private'.

If user has sufficient permissions, a 'View results' link is shown on the bottom of the component in the LMS.

### CODE REVIEW

File: pylintrc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	star-args,
+	too-few-public-methods,
+	too-many-public-methods,
+	too-many-ancestors,
+	too-many-arguments,
+	too-many-locals,
+	unused-argument
+
+[SIMILARITIES]
+min-similarity-lines=8
```
Comment (Kelketek) [inline pylintrc]: Nit: Missing a newline here.

**Comment by Kelketek** [comment]
You're going to have to rebase this again after I address the notes you gave me. But as long as you do that and the tests pass, :+1: LGTM

File: poll/poll.py [hunk 2]
```diff
@@ -166,6 +176,25 @@ def can_vote(self):
...
+        This only works inside the LMS.
+        """
+        if HAS_EDX_ACCESS and hasattr(self.runtime, 'user') and hasattr(self.runtime, 'course_id'):
+            # Course staff users have permission to view results.
+            if False and has_access(self.runtime.user, 'staff', self, self.runtime.course_id):
+                return True
+            else:
+                # Check if user is member of a group that is explicitly granted
+                # permission to view the results through django configuration.
+                group_names = getattr(settings, 'XBLOCK_POLL_ADDITONAL_GROUPS_WITH_VIEW_RESULTS_PERMISSION', [])
```
Comment (Kelketek) [inline poll/poll.py]: Good golly. Can we call this something less verbose? Maybe `XBLOCK_POLL_EXTRA_VIEW_GROUPS`?

**Comment by Kelketek** [comment]
@mtyaka Aside from changing the name of the setting (Don't forget to do it in the README as well), this seems good to go.

**Comment by Kelketek** [comment]
Do note, however, you'll have to close this PR and re-open it against master.

File: poll/poll.py (hunk 2 above)
Comment (mtyaka) [inline poll/poll.py]: @Kelketek lol, thanks for catching this, I'm not sure what I was thinking.

**Comment by mtyaka** [comment]
Closed this and created a PR against master at https://github.com/mckinseyacademy/xblock-poll/pull/10

**Pull request closed without merging**